components/ModEditSubmissionForm.vue

```diff
@@ -805,18 +808,20 @@ async function submitUpdatedSubmission(e: Event) {
             ]
         }
     }
-    try {
-        const result = await moderationSubmissionStore.updateSubmission(submissionInputVariables)
-        const submissionResult = result as { updateSubmission: Submission }
-        // This updates the submission in the form with the values stored in the db on success
-        initializeSubmissionFormValues(submissionResult.updateSubmission as Submission)
-        toast.success(t('modSubmissionForm.successMessageUpdated'))
-        if (moderationSubmissionStore.updatingSubmissionFromTopBar) {
-            router.push('/moderation')
-            moderationSubmissionStore.setUpdatingSubmissionFromTopBar(false)
-        }
-    } catch {
-        toast.error(t('modSubmissionForm.errorMessageUpdated'))
+
+    const result = await moderationSubmissionStore.updateSubmission(submissionInputVariables)
+    const submissionResult = result as ServerResponse<Mutation>
```

Comment:
let's set this as  a  return type of the `updateSubmission()`  function so the caller doesn't have to  set and figure out the types each time

Reply:
So `updateSubmission` is `Maybe<Submission>` so I will set it to `Submission` as the errors will be caught in the following part